net/connUDP.go

```diff
@@ -408,8 +513,18 @@ func (c *UDPConn) writeMulticast(ctx context.Context, raddr *net.UDPAddr, buffer
 	return nil
 }
 
+func (c *UDPConn) writeTo(raddr *net.UDPAddr, cm *ControlMessage, buffer []byte) (int, error) {
+	if !supportsOverrideRemoteAddr(c.connection) {
+		// If the remote address is set, we can use it as the destination address
+		// because the connection is already established.
+		// Note: Overwriting the destination address is only supported on Linux.
+		return c.connection.Write(buffer)
+	}
+	return c.packetConn.WriteTo(buffer, cm, raddr)
+}
+
 // WriteWithContext writes data with context.
-func (c *UDPConn) WriteWithContext(ctx context.Context, raddr *net.UDPAddr, buffer []byte) error {
+func (c *UDPConn) WriteWithContext(ctx context.Context, raddr *net.UDPAddr, cm *ControlMessage, buffer []byte) error {
```

Comment:
Is changing signatures `WriteWithContext` and `ReadWithContext` unproblematic although they are exported?

Reply:
It might break a few applications... easy to repair, but unexpected.